src/kOS/Function/Misc.cs

```diff
@@ -198,32 +181,20 @@ public override void Execute(SharedObjects shared)
                 shared.ScriptHandler.ClearContext("program");
                 //string filePath = shared.VolumeMgr.GetVolumeRawIdentifier(shared.VolumeMgr.CurrentVolume) + "/" + fileName;
                 var options = new CompilerOptions { LoadProgramsInSameAddressSpace = true, FuncManager = shared.FunctionManager };
-                var programContext = ((CPU)shared.Cpu).SwitchToProgramContext();
+                var programContext = shared.Cpu.SwitchToProgramContext();
 
                 List<CodePart> codeParts;
                 if (content.Category == FileCategory.KSM)
                 {
                     string prefix = programContext.Program.Count.ToString();
                     codeParts = content.AsParts(path, prefix);
+                    programContext.AddParts(codeParts);
+                    shared.Cpu.StopCompileStopwatch();
                 }
                 else
                 {
-                    try
-                    {
-                        codeParts = shared.ScriptHandler.Compile(path, 1, content.String, "program", options);
-                    }
-                    catch (Exception)
-                    {
-                        // If it died due to a compile error, then we won't really be able to switch to program context
-                        // as was implied by calling Cpu.SwitchToProgramContext() up above.  The CPU needs to be
-                        // told that it's still in interpreter context, or else it fails to advance the interpreter's
-                        // instruction pointer and it will just try the "call run()" instruction again:
```

Comment:
Oh now I see - The comment was cut and pasted from here.  Yeah it will need to be edited due to the fact that its not located here anymore.